src/app/lingua-puzzle/services/round/round.service.ts

```diff
@@ -0,0 +1,60 @@
+import { effect, inject, Injectable } from '@angular/core';
+import { Round } from '../../../shared/types/http-data.interface';
+import { HttpDataService } from '../../../core/services/http-data.service';
+import { Subject } from 'rxjs';
+import { LevelService } from '../level/level.service';
+
+enum CompletionStatus {
+  COMPLETED = 'completed',
+  PENDING = 'pending',
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RoundService {
+  private readonly httpData = inject(HttpDataService);
+  private readonly levelService = inject(LevelService);
+
+  public rounds: { round: Round; status: CompletionStatus }[] = [];
+  public round = new Subject<Round>();
+  public roundIndex = 0;
+
+  constructor() {
+    effect(() => {
+      this.levelService.currentLevel();
+      this.httpData.getRounds().subscribe((rounds) => {
+        this.rounds = rounds.map((round) => ({
+          round,
+          status: CompletionStatus.PENDING,
+        }));
+
+        this.setRound(0);
+      });
+    });
```

Comment:
This is very bad.

1. You are managing the state in effect,
2. You forgot to unsubscribe
3. You can make rounds reactive depended on httpData.getRounds()